custom/icds/location_reassignment/models.py

```diff
@@ -69,6 +74,20 @@ def valid(self):
                 self.errors.append("%s operation: location %s with site code %s is already deprecated." % (
                     self.type, new_location.name, new_location.site_code))
                 valid = False
+        valid = valid and self._validate_location_count(self.expected_old_locations,
+                                                        len(self.old_locations), "old")
+        valid = valid and self._validate_location_count(self.expected_new_locations,
+                                                        len(self.new_locations), "new")
+        return valid
+
+    def _validate_location_count(self, expected, count, location_type):
```

Comment:
I like this commit, it's a nice simplification and it's easy to read the new logic.